File: distributed-grep/DistributedGrep.scala
import com.twitter.scalding._

/** Grep! Distributed-ly!
  * 
  * Run this in local mode with:
  *   scald.rb --local DistributedGrep.scala
  */
class DistributedGrep(args : Args) extends Job(args) {

  val Pattern = ".*hello.*";

  Tsv("tweets.tsv", 'text)
    .filter('text) { text : String => text.matches(Pattern) }
    .write(Tsv("output.tsv"))    
}

File: distributed-grep/distributed_grep.clj
;;;; Distributed grep in Cascalog!

(def pattern #".*hello.*")

(deffilterop matches-pattern? [text pattern]
  (re-matches pattern text))

(defn distributed-grep [input pattern]
  (<- [?textline]
      (input ?textline)
      (matches-pattern? ?textline pattern)))

(?- (stdout) (distributed-grep (hfs-textline "tweets.tsv") pattern))